app/lib/features/pins/pages/pin_details_page.dart

```diff
@@ -0,0 +1,286 @@
+
+  // ignore: use_key_in_widget_constructors
+  const PinDetailsPage({
+    Key key = pinPageKey,
+    required this.pinId,
+  });
+
+  @override
+  ConsumerState<PinDetailsPage> createState() => _PinDetailsPageState();
+}
+
+class _PinDetailsPageState extends ConsumerState<PinDetailsPage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildAppBar(),
+      body: _buildBodyUI(),
+    );
+  }
+
+  AppBar _buildAppBar() {
+    return AppBar(
+      actions: [_buildActionMenu()],
+    );
+  }
+
+  Widget _buildBodyUI() {
+    final pinData = ref.watch(pinProvider(widget.pinId));
+    return pinData.when(
+      data: (pin) {
+        manageBackwardPinLinkSupport(context, ref, pin);
+        return SingleChildScrollView(
+          child: Column(
+            children: [
+              const SizedBox(height: 20),
+              _buildPinHeaderUI(pin),
+              const SizedBox(height: 20),
+              AttachmentSectionWidget(manager: pin.attachments()),
+              const SizedBox(height: 20),
+              CommentsSection(manager: pin.comments()),
+            ],
+          ),
+        );
+      },
+      loading: () => Skeletonizer(child: Text(L10n.of(context).loadingPin)),
+      error: (err, st) => Text(
+        L10n.of(context).errorLoadingPin(err),
+      ),
+    );
+  }
+
+  // pin actions menu builder
+  Widget _buildActionMenu() {
+    final pinData = ref.watch(pinProvider(widget.pinId));
+    return pinData.when(
+      data: (pin) {
+        return PopupMenuButton<String>(
+          key: PinDetailsPage.actionMenuKey,
+          icon: const Icon(Atlas.dots_vertical_thin),
+          itemBuilder: (context) => _buildAppBarActionMenuItems(pin),
+        );
+      },
+      loading: () => Skeletonizer(child: Text(L10n.of(context).loadingPin)),
+      error: (err, st) => const SizedBox.shrink(),
+    );
+  }
+
+  List<PopupMenuEntry<String>> _buildAppBarActionMenuItems(ActerPin pin) {
+    List<PopupMenuEntry<String>> actions = [];
+
+    //Get my membership details
+    final membership =
+        ref.watch(roomMembershipProvider(pin.roomIdStr())).valueOrNull;
+    if (membership != null) {
+      //Check for can post pin permission
+      if (membership.canString('CanPostPin')) {
+        //EDIT PIN TITLE MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            key: PinDetailsPage.editBtnKey,
+            onTap: () => showEditPintTitleDialog(context, ref, pin),
+            child: Text(L10n.of(context).editTitle),
+          ),
+        );
+
+        //EDIT PIN DESCRIPTION MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            key: PinDetailsPage.editBtnKey,
+            onTap: () => showEditPintDescriptionDialog(context, ref, pin),
+            child: Text(L10n.of(context).editDescription),
+          ),
+        );
+      }
+
+      final canRedact = ref.watch(canRedactProvider(pin));
+      if (canRedact.valueOrNull == true) {
```

Comment:
@gtalha07 @bitfriend @gnunicorn 
Any idea about getting `canRedact` always as `null` on this code?

Reply:
Not sure I'm following your question context? Care to elaborate?

Reply:
` if (canRedact.valueOrNull == true) {`
Above code return `false` as it is alway giving `null` data if though I have rights to do redact.

Reply:
code looks fine. are you admin of the space?

Reply:
Yes, I created pin and then tried to delete it but due to above condition delete option not showing.

Reply:
I have figure it out. Due to incorrect widget tree it was not rendering menu option again. Here is the commit which did correction on the same.
https://github.com/acterglobal/a3/commit/1f7fa6f5ae8c84fd08bd86d9b4aa16c79b7692b9